redisai/client.go

```diff
@@ -8,73 +8,136 @@ import (
 	"github.com/gomodule/redigo/redis"
 )
 
-// DeviceType is a device type
-type DeviceType string
+// Client is a RedisAI client
+type Client struct {
+	Pool *redis.Pool
+}
 
-// BackendType is a backend type
-type BackendType string
+func (c *Client) TensorGet(name string, ct TensorContentType) (data []interface{}, err error) {
+	args := redis.Args{}.Add(name, ct)
+	conn := c.Pool.Get()
+	defer conn.Close()
+	resp, err := conn.Do("AI.TENSORGET", args...)
+	data, err = processTensorReplyMeta(resp,err)
+	if ct == TensorContentTypeBlob {
+		data, err = processTensorReplyBlob(data,err)
+	}
+	if ct == TensorContentTypeValues {
+		data, err = processTensorReplyValues(data,err)
+	}
+	return
+}
 
-// DataType is a data type
-type DataType string
 
-const (
-	// BackendTF represents a TensorFlow backend
-	BackendTF = BackendType("TF")
-	// BackendTorch represents a Torch backend
-	BackendTorch = BackendType("TORCH")
-	// BackendONNX represents an ONNX backend
-	BackendONNX = BackendType("ORT")
+// TensorGetValues gets a tensor's values
+func (c *Client) TensorGetValues(name string) (dt DataType, shape []int, data interface{}, err error) {
+	resp,err := c.TensorGet(name,TensorContentTypeValues)
+	if err != nil {
+		return
+	}
+	return resp[0].(DataType), resp[1].([]int), resp[2], err
+}
 
-	// DeviceCPU represents a CPU device
-	DeviceCPU = DeviceType("CPU")
-	// DeviceGPU represents a GPU device
-	DeviceGPU = DeviceType("GPU")
+// TensorGetValues gets a tensor's values
+func (c *Client) TensorGetMeta(name string) (dt DataType, shape []int, err error) {
+	resp,err := c.TensorGet(name,TensorContentTypeMeta)
+	if err != nil {
```

Comment:
I think this will fail when the key doesn't exist - redigo returns a special error that you may want to handle in such cases instead of throwing it to the user (and having him/her need to include redigo specifically to handle this error).